librz/core/cmd/cmd_flag.c

```diff
@@ -779,620 +709,221 @@ RZ_IPI RzCmdStatus rz_flag_space_stack_list_handler(RzCore *core, int argc, cons
+			item = rz_flag_get(core->flags, argv[1] + 4);
 		}
-	} break;
-	case '-':
-		if (input[1] == '-') {
-			rz_flag_unset_all(core->flags);
-		} else if (input[1]) {
-			const char *flagname = rz_str_trim_head_ro(input + 1);
-			while (*flagname == ' ') {
-				flagname++;
-			}
-			if (*flagname == '.') {
-				RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, off, 0);
-				if (fcn) {
-					rz_analysis_function_delete_label_at(fcn, off);
-				} else {
-					eprintf("Cannot find function at 0x%08" PFMT64x "\n", off);
-				}
-			} else {
-				if (strchr(flagname, '*')) {
-					rz_flag_unset_glob(core->flags, flagname);
-				} else {
-					rz_flag_unset_name(core->flags, flagname);
-				}
-			}
-		} else {
-			rz_flag_unset_off(core->flags, off);
+		if (!item) {
+			RZ_LOG_ERROR("Cannot find flag '%s'\n", argv[1]);
+			return RZ_CMD_STATUS_ERROR;
 		}
-		break;
-	case '.': // "f."
-		input = rz_str_trim_head_ro(input + 1) - 1;
-		if (input[1]) {
-			if (input[1] == '*' || input[1] == 'j') {
-				if (input[2] == '*') {
-					print_function_labels(core->analysis, NULL, input[1]);
-				} else {
-					RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, off, 0);
-					if (fcn) {
-						print_function_labels(core->analysis, fcn, input[1]);
-					} else {
-						eprintf("Cannot find function at 0x%08" PFMT64x "\n", off);
-					}
-				}
-			} else {
-				char *name = strdup(input + ((input[2] == ' ') ? 2 : 1));
-				RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, off, 0);
-				if (name) {
-					char *eq = strchr(name, '=');
-					if (eq) {
-						*eq = 0;
-						off = rz_num_math(core->num, eq + 1);
-					}
-					rz_str_trim(name);
-					if (fcn) {
-						if (*name == '-') {
-							rz_analysis_function_delete_label(fcn, name + 1);
-						} else {
-							rz_analysis_function_set_label(fcn, name, off);
-						}
-					} else {
-						eprintf("Cannot find function at 0x%08" PFMT64x "\n", off);
-					}
-					free(name);
-				}
-			}
+		rz_flag_item_set_realname(item, argv[2]);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_flag_list_ascii_handler(RzCore *core, int argc, const char **argv) {
+	if (argc > 1) {
+		flagbars(core, argv[1]);
+	} else {
+		flagbars(core, NULL);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_flag_color_handler(RzCore *core, int argc, const char **argv) {
+	RzFlagItem *fi = rz_flag_get(core->flags, argv[1]);
+	if (!fi) {
+		RZ_LOG_ERROR("Cannot find the flag '%s'\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	const char *ret = rz_flag_item_set_color(fi, argv[2]);
+	if (ret) {
+		rz_cons_println(ret);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_flag_comment_handler(RzCore *core, int argc, const char **argv) {
+	RzFlagItem *item;
+	if (argc > 2) {
+		item = rz_flag_get(core->flags, argv[1]);
+		if (!item) {
+			RZ_LOG_ERROR("Cannot find flag with name '%s'\n", argv[1]);
+			return RZ_CMD_STATUS_ERROR;
+		}
+		if (!strncmp(argv[2], "base64:", 7)) {
```

Comment:
in theory we could easily implement a new "flag" in RzCmd RZ_CMD_ARG_FLAG_BASE64 similarly to RZ_CMD_ARG_FLAG_OPTION, RZ_CMD_ARG_FLAG_ARRAY, etc. so that if a arg is marked that way the parser can automatically decode it if it starts with `base64:` and the command handler just gets the decoded content without having to deal with decoding/encoding itself.

Reply:
Probably, yes, but could be done separately, IMHO.

Reply:
absolutely. Just wanted to know what you think about this, whether you have seen this spread in many places... Because if it's just 2/3 places, maybe it's fine as is.

Reply:
I don't think it's used often, probably doesn't make sense to make a separate argument type for this. I think having `@base64:` would make more sense instead.